### PR TITLE
adder: modify syntax tree instead of strings

### PR DESCRIPTION
**Pull request by ryantm**
Why
===
* nix-editor was doing string manipulation inside of adder on the entire contents of the document.
* With this commit, now splice the syntax tree, and only allow editing the deps_list part passed in, instead of the whole tree.
* Also, use the earlier whitespace in the document as a clue for how deeply to indent the entries, which fixes and normalizes a variety of edge cases.
* The adder test cases had some redundant code paths.

What changed
===
* In verify_get, also record the last whitespace we've seen with a newline in it. Then use this to determine what level to indent addtional entries.
* Modify verify_get to take the tree as a borrow, because we need to have the deps_list be referencing the same memory as the tree, so that add can modify the tree datastructure in place later.
* Make a new SyntaxNodeAndWhitespace struct to store the node next to the whitespace. The whitespace is optional because there might not be any.
* Change add_dep to  no longer pass in  the contents of the  file as a string. Also, change  it to return a modified SyntaxNode  instead of a String.
* Use splice_children to change the SyntaxNode
* Remove clowny indenting code and use the Whitespace passed in, if available.
* Add more tests to verify indenting
* Simplify the adder tests for Python entry adds to use the same test helper as the regular adds

Test plan
===
* nix develop
* cargo test

### CODE REVIEW

**Comment by masad-frost** [comment]
@cdmistman this one goes to you

File: src/adder.rs [hunk 1]
```diff
@@ -61,62 +55,85 @@ mod add_tests {
...
     pkgs.test
   ];
 }
         "#)
     }
 
-    fn python_replit_nix() -> String {
-        r#"
-{ pkgs }: {
+    fn python_replit_nix() -> &'static str {
```
Comment (cdmistman) [inline src/adder.rs]: can you make this `const` instead

Comment (ryantm) [reply src/adder.rs]: Thanks, fixed!

File: src/adder.rs [hunk 2]
```diff
@@ -1,58 +1,52 @@
...
 
-    // we need to add leading whitespace to the next line so that
-    // the pkgs are correctly formatted (i.e. they are lined up)
-    let white_space_count = if add_dep_pos >= 2 + open_bracket_pos {
-        add_dep_pos - open_bracket_pos - 2
-    } else {
-        2
+    let newline =  match has_newline {
+        true => "",
+        false => newline_whitespace,
```
Comment (cdmistman) [inline src/adder.rs]: can you check for the newline first and then determine if you need `\n` at the beginning of the string?

Comment (cdmistman) [reply src/adder.rs]: oh you don't need `base_whitespace` later either? instead can you put all the string construction logic inside of the `false` match arm?

**Review by cdmistman — APPROVED** [review]
